testsuite/tests/kuadrant/authorino/operator/tls/test_webhook.py

```diff
@@ -0,0 +1,133 @@
+    """Certificate hierarchy used for the tests.
+    Authorino certificate has *hosts* set to *authorino_domain* value.
+    """
+    chain = {
+        "envoy_ca": CertInfo(children={
+            "envoy_cert": None,
+            "valid_cert": None
+        }),
+        "authorino_ca": CertInfo(children={
+            "authorino_cert": CertInfo(hosts=authorino_domain),
+        }),
+        "invalid_ca": CertInfo(children={
+            "invalid_cert": None
+        })
+    }
+    return cert_builder(cfssl, chain, wildcard_domain)
+
+
+@pytest.fixture(scope="module")
+def authorino_parameters(authorino_parameters, specific_authorino_name):
+    """Setup TLS with specific name for authorino."""
+    authorino_parameters['name'] = specific_authorino_name
+    return authorino_parameters
+
+
+# pylint: disable=unused-argument
+@pytest.fixture(scope="module")
+def authorization(authorization, openshift, module_label, authorino_domain) -> Authorization:
+    """In case of Authorino, AuthConfig used for authorization"""
+    authorization.remove_all_hosts()
+    authorization.add_host(authorino_domain)
+
+    authorization.identity.remove_all()
+    plain = {'authJSON': 'context.request.http.body.@fromstr|request.userInfo'}
+    authorization.identity.kubernetes('k8s-userinfo', plain)
+
+
+    authorization.authorization.opa_policy('features', OPA_POLICY)
+
+    when = [{'selector': 'auth.authorization.features.allow', 'operator': 'eq', 'value': 'true'},
+            {'selector': 'auth.authorization.features.verb', 'operator': 'eq', 'value': 'CREATE'}]
+    kube_user = {'valueFrom': {'authJSON': 'auth.identity.username'}}
+    kube_attrs = {
+                    'namespace': {'value': openshift.project},
+                    'group': {'value': 'networking.k8s.io'},
+                    'resources': {'value': 'Ingress'},
+                    'verb': {'value': 'create'}
+                 }
+    authorization.authorization.kubernetes('ingress-authn-k8s-binding-create',
+                                           when, kube_user, kube_attrs, 1)
+
+    when = [{'selector': 'auth.authorization.features.allow', 'operator': 'eq', 'value': 'true'},
+            {'selector': 'auth.authorization.features.verb', 'operator': 'eq', 'value': 'DELETE'}]
+    kube_user = {'valueFrom': {'authJSON': 'auth.identity.username'}}
+    kube_attrs = {
+                    'namespace': {'value': openshift.project},
+                    'group': {'value': 'networking.k8s.io'},
+                    'resources': {'value': 'Ingress'},
+                    'verb': {'value': 'delete'}
+                 }
+    authorization.authorization.kubernetes('ingress-authn-k8s-binding-delete',
+                                           when, kube_user, kube_attrs, 1)
+    return authorization
+
+
+@pytest.fixture(scope="module")
+def validating_webhook(openshift, authorino_domain, certificates):
+    """Create validating webhook."""
+    name = 'check-ingress.authorino.kuadrant.io'
```

Comment:
What will happen if two people run this test in parallel? Wouldnt it break because the name is not randomised?